eZ/Publish/Core/Persistence/Cache/TrashHandler.php

```diff
@@ -99,7 +99,25 @@ public function findTrashItems(Criterion $criterion = null, $offset = 0, $limit
     public function emptyTrash()
     {
         $this->logger->logCall(__METHOD__, array());
+
+        $trashedItems = $this->persistenceHandler->trashHandler()->findTrashItems();
+
+        $tags = [[]];
+        foreach ($trashedItems as $trashedItem) {
+            $contentId = $trashedItem->contentId;
+            $reverseRelations = $this->persistenceHandler->contentHandler()->loadReverseRelations($contentId);
+
+            $tags[] = array_map(function (Relation $relation) {
```

Comment:
You can avoid array_merge by replacing array_map with array_walk (or just inner loop).